src/test/compile-fail/impl-duplicate-methods.rs

```diff
@@ -11,7 +11,7 @@
 struct Foo;
 impl Foo {
     fn orange(&self){}
-    fn orange(&self){}   //~ ERROR error: duplicate method
+    fn orange(&self){}   //~ ERROR duplicate associated function
```

Comment:
technically, methods are also associated functions, but we have been
calling associated functions that take self methods

On Wed, Jun 3, 2015 at 7:19 AM, Nick Hamann <notifications@github.com>
wrote:

> In src/test/compile-fail/impl-duplicate-methods.rs
> <https://github.com/rust-lang/rust/pull/25981#discussion_r31594420>:
>
> > @@ -11,7 +11,7 @@
> >  struct Foo;
> >  impl Foo {
> >      fn orange(&self){}
> > -    fn orange(&self){}   //~ ERROR error: duplicate method
> > +    fn orange(&self){}   //~ ERROR duplicate associated function
>
> Ahh, so maybe what this message should really do is say "duplicate method"
> when we're dealing with methods, and "duplicate associated function"
> otherwise. I will see if I can make that work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/25981/files#r31594420>.
>